src/compilation/mod.rs

```diff
@@ -31,31 +36,37 @@ impl<'a> JSONSchema<'a> {
     /// Compile the input schema into a validation tree
     pub fn compile(
         schema: &'a Value,
-        draft: Option<schemas::Draft>,
+        config: Option<JSONSchemaCompileConfig>,
     ) -> Result<JSONSchema<'a>, CompilationError> {
         // Draft is detected in the following precedence order:
         //   - Explicitly specified;
         //   - $schema field in the document;
         //   - Draft7;
-        let draft = draft.unwrap_or_else(|| {
-            schemas::draft_from_schema(schema).unwrap_or(schemas::Draft::Draft7)
-        });
+
+        let mut config = config.unwrap_or_default();
+        config.set_draft_if_missing(schema);
```

Comment:
Maybe it will be better to disallow creating `JSONSchemaCompileConfig` without `draft`